storey/table.py

```diff
@@ -244,6 +288,151 @@ def __getitem__(self, key):
+            if self._flush_task:
+                self._flush_task.cancel()
+                with suppress(asyncio.CancelledError):
+                    await self._flush_task
+            self._q = None
+            self._flush_task = None
+            self._flush_exception = None
+
+    async def _persist(self, job, from_terminate=False):
+        if self._flush_exception is not None:
+            raise self._flush_exception
+        if not self._q:
+            self._q = asyncio.queues.Queue(self._max_updates_in_flight)
+            self._worker_awaitable = asyncio.get_running_loop().create_task(self._persist_worker())
+            if not from_terminate:
+                # for flow reuse
+                self._terminated = False
+
+        if self._worker_awaitable.done():
+            await self._worker_awaitable
+            raise FlowError("Persist worker has already terminated")
+        else:
+            # Initializing the key with 2 lists. One for pending requests and one for requests that an update request has been issued for.
+            if isinstance(job.key, list):
+                # list can't be key in a dictionary
+                job_key = str(job.key)
+            else:
+                job_key = job.key
+            if job_key not in self._pending_by_key:
+                self._pending_by_key[job_key] = _PendingEvent()
+
+            # If there is a current update in flight for the key, add the event to the pending list. Otherwise update the key.
+            self._pending_by_key[job_key].pending.append(job)
+            if len(self._pending_by_key[job_key].in_flight) == 0:
+                self._pending_by_key[job_key].in_flight = self._pending_by_key[job_key].pending
+                self._pending_by_key[job_key].pending = []
+                task = self._safe_process_events(self._pending_by_key[job_key].in_flight)
+                await self._q.put((job, asyncio.get_running_loop().create_task(task)))
+                if self._worker_awaitable.done():
+                    await self._worker_awaitable
+
+    async def _safe_process_events(self, jobs):
+        try:
+            # TODO using only last event might not work correctly if
+            #  there are different non aggregation attrs in each event
+            job = jobs[-1]
+            if isinstance(job.key, list):
+                # list can't be key in a dictionary
+                job_key = str(job.key)
+            else:
+                job_key = job.key
+            return await self._internal_persist_key(job_key, job.data)
+        except BaseException as ex:
+            for job in jobs:
+                if job.extra_data and job.extra_data._awaitable_result:
+                    none_or_coroutine = job.extra_data._awaitable_result._set_error(ex)
+                    if none_or_coroutine:
+                        await none_or_coroutine
+            raise ex
+
+    def update_interval(self, flush_interval: FlushInterval):
```

Comment:
This method is never called.

Reply:
orit asked for the interval to be configurable, i added this method to be able to change flush interval in runtime without stopping the process

Reply:
I'm not sure I understand in what scenarios this API can be used.